go/consensus/cometbft/apps/vault/vault.go

```diff
@@ -0,0 +1,124 @@
+package vault
+
+import (
+	"github.com/cometbft/cometbft/abci/types"
+
+	"github.com/oasisprotocol/oasis-core/go/common/cbor"
+	"github.com/oasisprotocol/oasis-core/go/consensus/api/transaction"
+	"github.com/oasisprotocol/oasis-core/go/consensus/cometbft/api"
+	governanceApi "github.com/oasisprotocol/oasis-core/go/consensus/cometbft/apps/governance/api"
+	stakingapp "github.com/oasisprotocol/oasis-core/go/consensus/cometbft/apps/staking"
+	stakingApi "github.com/oasisprotocol/oasis-core/go/consensus/cometbft/apps/staking/api"
+	vaultState "github.com/oasisprotocol/oasis-core/go/consensus/cometbft/apps/vault/state"
+	vault "github.com/oasisprotocol/oasis-core/go/vault/api"
+)
+
+var (
+	_ api.Application          = (*vaultApplication)(nil)
+	_ api.TogglableApplication = (*vaultApplication)(nil)
+)
+
+type vaultApplication struct {
+	state api.ApplicationState
+	md    api.MessageDispatcher
+}
+
+func (app *vaultApplication) Name() string {
+	return AppName
+}
+
+func (app *vaultApplication) ID() uint8 {
+	return AppID
+}
+
+func (app *vaultApplication) Methods() []transaction.MethodName {
+	return vault.Methods
+}
+
+// Enabled implements api.TogglableApplication and api.TogglableMessageSubscriber.
+func (app *vaultApplication) Enabled(ctx *api.Context) (bool, error) {
```

Comment:
I asked already, but will ask again. Is it possible to remove returned `error`, update `ConsensusParameters` to return a special error if parameters are not set, handle this error here and return `false` in that case?

Reply:
But the error could also be a "state unavailable" error which needs to be propagated. If we get rid of the `error` return, then such propagation would not be possible or we would need to panic here which is not ideal as the caller should decide how to handle that.

Reply:
Ok. Resolving.